src/main/scala/io/lambdaworks/workshop/recursion/NonTail2TailRecursion.scala

```diff
@@ -1,19 +1,31 @@
 package io.lambdaworks.workshop.recursion
 
+import scala.annotation.tailrec
+
 /**
   * Rewrite below non tail-recursive functions to tail-recursive one.
   * Add @tailrec annotation to prove it.
   */
 object NonTail2TailRecursion {
 
-  def factorial(n: Int): Int =
-    if (n <= 0) 1 else n * factorial(n - 1)
+  def factorial(n: Int): Int = {
+    @tailrec
+    def factorialRec(n: Int, acc: Int = 1): Int = {
+      if (n <= 0) acc else factorialRec(n - 1, n * acc)
+    }
```

Comment:
You can omit curly brackets.

```suggestion
    def factorialRec(n: Int, acc: Int = 1): Int = 
      if (n <= 0) acc else factorialRec(n - 1, n * acc)

```